relay-server/src/services/buffer/envelope_stack/sqlite.rs

```diff
@@ -226,6 +226,10 @@ impl EnvelopeStack for SqliteEnvelopeStack {
 
         Ok(result)
     }
+
+    fn drain(self) -> Vec<Box<Envelope>> {
```

Comment:
nit: The total number of in-memory envelopes is probably low, but I anyway wonder whether we could return an iterator here, so we don't have to allocate an extra vector.